pkg/kubelet/kubelet_pods.go

```diff
@@ -743,6 +743,20 @@ func (kl *Kubelet) podIsTerminated(pod *v1.Pod) bool {
 	return status.Phase == v1.PodFailed || status.Phase == v1.PodSucceeded || (pod.DeletionTimestamp != nil && notRunning(status.ContainerStatuses))
 }
 
+// podIsFinished returns true if pod is in the finished state ("Failed" or "Succeeded").
+func (kl *Kubelet) podIsFinished(pod *v1.Pod) bool {
```

Comment:
nit: I am not sure about `podIsFinished`. It is kind of confusing that we have both "podIsTerminated" and "podIsFinished". I don't have any better suggestion though.